src/H5Tconv_float.c

```diff
@@ -688,602 +755,628 @@ H5T__conv_f_i(const H5T_t *src_p, const H5T_t *dst_p, H5T_cdata_t *cdata, const
             if (NULL == conv_ctx)
                 HGOTO_ERROR(H5E_ARGS, H5E_BADVALUE, FAIL, "invalid datatype conversion context pointer");
 
-            src = src_p->shared->u.atomic;
```

Comment:
All the conversion logic extracted to `H5T__conv_f_i_loop` helper function for reuse with complex numbers, with just an addition to do the byte-swapping correctly for complex numbers.